internal/tests/pytests/util/run_util/test_run_util.py

```diff
@@ -49,52 +85,322 @@ def test_pre_run_setup_env_vars():
+    config.set('config', 'PROCESS_LIST', 'GridStat,Usage')
+    actual = ru.run_metplus(config)
+    assert actual == 0
+    out, err = capfd.readouterr()
+    assert "USAGE: This text is displayed when [config] PROCESS_LIST = Usage." in out
+
+
+@pytest.mark.parametrize(
+    "config_dict,expected,check_err",
+    [
+        (
+            # Check init errors raised
+            {'PROCESS_LIST': 'GridStat, PCPCombine'},
+            4,
+            [
+                'Must set either FCST_PCP_COMBINE_RUN or OBS_PCP_COMBINE_RUN',
+                'OBS_GRID_STAT_INPUT_TEMPLATE required to run',
+            ],
+        ),
+        (
+            # Check correct config runs without error
+            {
+                'PROCESS_LIST': 'GridStat',
+                'LOOP_BY': 'INIT',
+                'INIT_TIME_FMT': '%Y%m%d%H',
+                'INIT_BEG': '2005080700',
+                'INIT_END': '2005080700',
+                'INIT_INCREMENT': '12H',
+                'FCST_GRID_STAT_INPUT_DIR': '{INPUT_BASE}/met_test/data/sample_fcst',
+                'FCST_GRID_STAT_INPUT_TEMPLATE': '{init?fmt=%Y%m%d%H}/wrfprs_ruc13_{lead?fmt=%HH}.tm00_G212',
+                'OBS_GRID_STAT_INPUT_DIR': '{INPUT_BASE}/met_test/new',
+                'OBS_GRID_STAT_INPUT_TEMPLATE': 'ST2ml{valid?fmt=%Y%m%d%H}_A03h.nc',
+                'GRID_STAT_OUTPUT_DIR': '{OUTPUT_BASE}/met_tool_wrapper/GridStat/GridStat',
+                'GRID_STAT_OUTPUT_TEMPLATE': '{init?fmt=%Y%m%d%H}',
+                'BOTH_VAR1_NAME': 'APCP',
+                'BOTH_VAR1_LEVELS': 'A03',
+                'BOTH_VAR1_THRESH': 'gt1, gt5',
+                'INPUT_MUST_EXIST': 'False',
+            },
+            0,
+            None,
+        ),
+        # More config examples can be added here.
+    ],
+)
+@pytest.mark.util
+def test_run_metplus(capfd, config_dict, expected, check_err):
+    config = get_config_from_file()
+    for k, v in config_dict.items():
+        config.set('config', k, v)
+
+    actual = ru.run_metplus(config)
+    assert actual == expected
+
     out, err = capfd.readouterr()
-    assert 'MET_INSTALL_DIR must be set correctly to run METplus' in err 
+    if check_err:
+        for msg in check_err:
+            assert msg in err
+    else:
+        assert err == ''
+
+
+@pytest.mark.parametrize(
+    "side_effect,return_value,check_err",
+    [
+        (Exception, None, "Fatal error occurred"),
+        (None, [], None),
+    ],
+)
+@pytest.mark.util
+def test_run_metplus_errors(capfd, side_effect, return_value, check_err):
+    with mock.patch.object(
+        ru, '_load_all_wrappers', side_effect=side_effect, return_value=return_value
+    ):
+        config = get_config_from_file()
+        config.set('config', 'PROCESS_LIST', 'GridStat')
+        actual = ru.run_metplus(config)
+        assert actual == 1
+        if check_err:
+            out, err = capfd.readouterr()
+            assert "Fatal error occurred" in err
+        else:
+            err = ''
```

Comment:
Is this supposed to include an assert? If so, would it make sense to change the 2nd check_err value to an empty string instead of None, then move the assert in the if statement after the if/else statement and change it to:

`assert check_err == err`

Reply:
Oops, a bit sloppy from me here, sorry. Fixed in latest commit.